### PR TITLE
MODSOURCE-567 Link update report: send event if update failed

### PR DESCRIPTION
**Pull request by Shans-Kaluhin**
When SRS consumes an event from `links.instance-authority` it tries to update related SRS records. If the update failed then send an event to `links.instance-authority-stats` topic.

https://issues.folio.org/browse/MODSOURCE-567

### CODE REVIEW

**Review by hasanepam — APPROVED** [review]

**Comment by Shans-Kaluhin** [comment]
This PR waits for merging this PR's:
`ramls`: https://github.com/folio-org/data-import-raml-storage/pull/271
`folio-kafka-wrapper`: https://github.com/folio-org/folio-kafka-wrapper/pull/35

**Review by hasanepam — APPROVED** [review]

File: mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityLinkChunkKafkaHandler.java [hunk 1]
```diff
@@ -262,49 +293,58 @@ private Future<BibAuthorityLinksUpdate> createSnapshot(BibAuthorityLinksUpdate b
...
       return Future.failedFuture(e);
     }
-    return promise.future();
+    return promise.future().onFailure(th -> LOGGER.error("Failed to send {} event for jobId {}.", topic.topicName(), jobId, th));
   }
 
-  private KafkaProducerRecord<String, String> createKafkaProducerRecord(MarcBibUpdate marcBibUpdate,
+  private KafkaProducerRecord<String, String> createKafkaProducerRecord(KafkaTopic topic, String tenant, Object record,
                                                                         List<KafkaHeader> kafkaHeaders) {
-    var topicName = createTopicNameNoNamespace(SRS_BIB_UPDATE_TOPIC, marcBibUpdate.getTenant(), kafkaConfig);
```
Comment (viacheslavkol) [inline mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityLinkChunkKafkaHandler.java:305]: This method could be also removed from utils

File: mod-source-record-storage-server/src/test/java/org/folio/services/AuthorityLinkChunkKafkaHandlerTest.java [hunk 2]
```diff
@@ -63,32 +67,34 @@ public class AuthorityLinkChunkKafkaHandlerTest extends AbstractLBServiceTest {
   private static final String PARSED_MARC_RECORD_UNLINKED = "src/test/resources/parsedMarcRecordUnlinked.json";
   private static final String KAFKA_KEY_NAME = "test-key";
   private static final String KAFKA_TEST_HEADER = "test";
-  private static final String KAFKA_CONSUMER_TOPIC = getTopicName(AUTHORITY_INSTANCE_LINKS_TOPIC);
-  private static final String KAFKA_PRODUCER_TOPIC = getTopicName(SRS_BIB_UPDATE_TOPIC);
+  private static final String KAFKA_CONSUMER_TOPIC = getTopicName(INSTANCE_AUTHORITY);
+  private static final String KAFKA_KAFKA_SRS_BIB_PRODUCER_TOPIC = getTopicName(MARC_BIB);
```
Comment (viacheslavkol) [inline mod-source-record-storage-server/src/test/java/org/folio/services/AuthorityLinkChunkKafkaHandlerTest.java:71]: Naming trouble

**Review by viacheslavkol — APPROVED** [review]

**Review by psmagin — CHANGES_REQUESTED** [review]
I don't think that we should send success events from srs. It's like a completion event for the whole update, and it can be completed fully only in inventory.

**Pull request merged**